library/std/src/sys/unix/fd.rs

```diff
@@ -26,6 +27,31 @@ const READ_LIMIT: usize = c_int::MAX as usize - 1;
 #[cfg(not(target_os = "macos"))]
 const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
+#[cfg(any(target_os = "linux", target_os = "macos"))]
+fn max_iov() -> c_int {
+    static LIM: SyncOnceCell<c_int> = SyncOnceCell::new();
```

Comment:
Drive by comment, but for things that don't need exactly once semantics I prefer code like this, which doesn't invoke synchronization machinery:

```
static CACHE: AtomicUsize = AtomicUsize::new(!0);
let val = CACHE.load(Relaxed);
if val != !0 { return val }
let val = compute_val();
CACHE.store(val, Relaxed)
val
```

There's no real benefits to this, just using the smallest tool for the job.

Reply:
Thank you for the advice! I adjusted the code following your suggestion but using zero as the initial value as this is not a valid value in this case.